src/components/FetchingData.jsx

```diff
@@ -0,0 +1,35 @@
+import useFetch from "./useFetch";
+import Loading from './Loading';
+import { Link } from "react-router-dom";
+import '../App.css';
+
+const FetchingData = () => {
+  const { data: products, loading, error} = useFetch('products');
+
+  if (error) {
+    return <p>An error occurred</p>;
+  }
+
+  if (loading) {
+    return <Loading />;
+  }
+
+  return (
+    <div className="product-container">
+      {products.length > 0 ? (
+        products.map((product) => (
+          <div className="product-item" key={product.id}>
```

Comment:
I would have liked to have seen you reuse the components you made earlier in the course. For example, this would be a `Box` component.